keras_nlp/models/generative_task.py

```diff
@@ -105,7 +106,7 @@ def compiled_generate_function(inputs, end_token_id, state):
 
             def wrapped_generate_function(
```

Comment:
Since we have this wrapper here, why don't we force a tuple here? If list -> cast tuple. Good to remove backend specific gotchas like this.